pkg/app/api/service/webservice/service.proto

```diff
@@ -442,15 +444,25 @@ message ListAPIKeysResponse {
     repeated model.APIKey keys = 1;
 }
 
-message GetInsightDataRequest {
+message GetInsightMetricsDataRequest {
     pipe.model.InsightMetricsKind metrics_kind = 1 [(validate.rules).enum.defined_only = true];
     pipe.model.InsightStep step = 2 [(validate.rules).enum.defined_only = true];
     int64 range_from = 3 [(validate.rules).int64.gt = 0];
     int64 data_point_count = 4 [(validate.rules).int64.gt = 0];
     string application_id = 5;
 }
 
-message GetInsightDataResponse {
+message GetInsightDataPointsResponse {
     int64 updated_at = 1;
     repeated pipe.model.InsightDataPoint data_points = 2;
 }
+
+message GetInsightPieChartDataRequest {
+    pipe.model.PieChartKind kind = 1 [(validate.rules).enum.defined_only = true];
+    string project_id = 2 [(validate.rules).string.min_len = 1];
```

Comment:
We don't need the projectID because it is already included in the JWT token used while authenticating.